var/spack/repos/builtin/packages/rdma-core/package.py

```diff
@@ -69,6 +70,8 @@ class RdmaCore(CMakePackage):
     version("17.1", sha256="b47444b7c05d3906deb8771eec3e634984dd83f5e620d5e37d3a83f74f0cc1ba")
     version("13", sha256="e5230fd7cda610753ad1252b40a28b1e9cf836423a10d8c2525b081527760d97")
 
+    patch("libdrm.patch")
```

Comment:
Shouldn't this only be applied to `@34:`? Also, since this is arch specific, shouldn't the patch only be applied in archlinux?

Reply:
This patch comes directly from the archlinux rdma-core package. Without it, there is no chance to install it in any Archlinux-related distro that installs the DRM code in the canonical place as instructed in its package. This is the `libdrm` packaging/installation script in archlinux:

```
# Maintainer: Andreas Radke <andyrtr@archlinux.org>

pkgname=libdrm
pkgver=2.4.121
pkgrel=1
pkgdesc="Userspace interface to kernel DRM services"
url="https://dri.freedesktop.org/"
arch=(x86_64)
license=('MIT')
depends=('libpciaccess' 'glibc')
makedepends=('python-docutils' 'meson')
checkdepends=('cairo')
optdepends=('cairo: needed for modetest tool')
replaces=('libdrm-new' 'libdrm-nouveau')
source=(https://dri.freedesktop.org/$pkgname/$pkgname-$pkgver.tar.xz
        COPYING)
sha512sums=('cc8816d61884caa0e404348d1caeb0b2952fb50e1dc401716adfe08121096e2a67826db0bda0d8b163d67c5ee048870177670d5eac28a5abe5792d09ba77ab2e'
            'b0ca349b882a4326b19f81f22804fabdb6fb7aef31cdc7b16b0a7ae191bfbb50c7daddb2fc4e6c33f1136af06d060a273de36f6f3412ea326f16fa4309fda660')

build() {
  arch-meson $pkgname-$pkgver build \
    -D udev=true \
    -D etnaviv=disabled \
    -D freedreno=disabled \
    -D vc4=disabled \
    -D valgrind=disabled \
    -D install-test-programs=true
  meson compile -C build
}

check() {
  meson test -C build --print-errorlogs
}

package() {
  meson install -C build --destdir "$pkgdir"
  install -Dt "$pkgdir/usr/share/licenses/$pkgname" -m644 COPYING
}   
```

This installation results without any redirection of paths to the following tree:

```
.
└── usr
    ├── bin
    │   ├── amdgpu_stress
    │   ├── drmdevice
    │   ├── modeprint
    │   ├── modetest
    │   ├── proptest
    │   └── vbltest
    ├── include
    │   ├── libdrm
    │   │   ├── amdgpu_drm.h
    │   │   ├── amdgpu.h
    │   │   ├── drm_fourcc.h
    │   │   ├── drm.h
    │   │   ├── drm_mode.h
    │   │   ├── drm_sarea.h
    │   │   ├── i915_drm.h
    │   │   ├── intel_aub.h
    │   │   ├── intel_bufmgr.h
    │   │   ├── intel_debug.h
    │   │   ├── mach64_drm.h
    │   │   ├── mga_drm.h
    │   │   ├── msm_drm.h
    │   │   ├── nouveau
    │   │   │   ├── nouveau.h
    │   │   │   └── nvif
    │   │   │       ├── cl0080.h
    │   │   │       ├── cl9097.h
    │   │   │       ├── class.h
    │   │   │       ├── if0002.h
    │   │   │       ├── if0003.h
    │   │   │       ├── ioctl.h
    │   │   │       └── unpack.h
    │   │   ├── nouveau_drm.h
    │   │   ├── qxl_drm.h
    │   │   ├── r128_drm.h
    │   │   ├── r600_pci_ids.h
    │   │   ├── radeon_bo_gem.h
    │   │   ├── radeon_bo.h
    │   │   ├── radeon_bo_int.h
    │   │   ├── radeon_cs_gem.h
    │   │   ├── radeon_cs.h
    │   │   ├── radeon_cs_int.h
    │   │   ├── radeon_drm.h
    │   │   ├── radeon_surface.h
    │   │   ├── savage_drm.h
    │   │   ├── sis_drm.h
    │   │   ├── tegra_drm.h
    │   │   ├── vc4_drm.h
    │   │   ├── via_drm.h
    │   │   ├── virtgpu_drm.h
    │   │   └── vmwgfx_drm.h
    │   ├── libsync.h
    │   ├── xf86drm.h
    │   └── xf86drmMode.h
    ├── lib
    │   ├── libdrm_amdgpu.so -> libdrm_amdgpu.so.1
    │   ├── libdrm_amdgpu.so.1 -> libdrm_amdgpu.so.1.0.0
    │   ├── libdrm_amdgpu.so.1.0.0
    │   ├── libdrm_intel.so -> libdrm_intel.so.1
    │   ├── libdrm_intel.so.1 -> libdrm_intel.so.1.0.0
    │   ├── libdrm_intel.so.1.0.0
    │   ├── libdrm_nouveau.so -> libdrm_nouveau.so.2
    │   ├── libdrm_nouveau.so.2 -> libdrm_nouveau.so.2.0.0
    │   ├── libdrm_nouveau.so.2.0.0
    │   ├── libdrm_radeon.so -> libdrm_radeon.so.1
    │   ├── libdrm_radeon.so.1 -> libdrm_radeon.so.1.0.1
    │   ├── libdrm_radeon.so.1.0.1
    │   ├── libdrm.so -> libdrm.so.2
    │   ├── libdrm.so.2 -> libdrm.so.2.4.0
    │   ├── libdrm.so.2.4.0
    │   └── pkgconfig
    │       ├── libdrm_amdgpu.pc
    │       ├── libdrm_intel.pc
    │       ├── libdrm_nouveau.pc
    │       ├── libdrm.pc
    │       └── libdrm_radeon.pc
    └── share
        ├── libdrm
        │   └── amdgpu.ids
        ├── licenses
        │   └── libdrm
        │       └── COPYING
        └── man
            ├── man3
            │   ├── drmAvailable.3.gz
            │   ├── drmHandleEvent.3.gz
            │   └── drmModeGetResources.3.gz
            └── man7
                ├── drm.7.gz
                ├── drm-kms.7.gz
                └── drm-memory.7.gz

16 directories, 77 files
``` 

So, I suspect that Redhat variants have unilaterally changed the path to `include/drm`, and the `rdma-core` cmake script has been designed mostly with those distros in mind.


This is a small patch with zero repercussions for others, so it would be great if you could consider adding it. Although I understand it is not the ideal solution.

Reply:
I sent a fix upstream, but in the best-case scenario will only be available for future versions:

https://github.com/linux-rdma/rdma-core/pull/1477

Reply:
> Shouldn't this only be applied to `@34:`? Also, since this is archlinux specific, shouldn't the patch only be applied in archlinux?

We've made it so the code is platform agnostic

Reply:
> Shouldn't this only be applied to `@34:`? Also, since this is archlinux specific, shouldn't the patch only be applied in archlinux?

yes